python/ray/serve/_private/utils.py

```diff
@@ -628,10 +623,17 @@ def send_forever():
                         logger.warning(
                             f"MetricsPusher thread failed to run metric task: {e}"
                         )
-                duration_s = time.time() - start
-                remaining_time = least_interval_s - duration_s
-                if remaining_time > 0:
-                    time.sleep(remaining_time)
+
+                # For all tasks, check when the task should be executed
+                # next. Sleep until the next closest time.
+                least_interval_s = math.inf
```

Comment:
[Nit] Could we raise an error if the `MetricsPusher` is started without any tasks registered? As written, it silently sleeps forever since `least_interval_s` gets set to `math.inf`.

Reply:
That makes sense! I've added this `MetricsPusher.start()`